src/tree.rs

```diff
@@ -625,6 +627,128 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ```
+    /// use id_tree::Tree;
+    /// use id_tree::Node;
+    ///
+    /// let root_node = Node::new(1);
+    /// let first_child_node = Node::new(2);
+    /// let second_child_node = Node::new(3);
+    /// let grandchild_node = Node::new(4);
+    ///
+    /// let mut tree: Tree<i32> = Tree::new();
+    /// let root_id = tree.set_root(root_node);
+    ///
+    /// let first_child_id  = tree.insert_with_parent(first_child_node,  &root_id).ok().unwrap();
+    /// let second_child_id = tree.insert_with_parent(second_child_node, &root_id).ok().unwrap();
+    /// let grandchild_id   = tree.insert_with_parent(grandchild_node, &second_child_id).unwrap();
+    ///
+    /// tree.swap_sub_tree(&first_child_id, &grandchild_id).unwrap();
+    ///
+    /// assert!(tree.get(&second_child_id).unwrap().children().contains(&first_child_id));
+    /// assert!(tree.get(&root_id).unwrap().children().contains(&grandchild_id));
+    /// ```
+    ///
+    pub fn swap_sub_tree(&mut self, first_id: &NodeId, second_id: &NodeId) -> Result<(), NodeIdError> {
+        let (is_valid, error) = self.is_valid_node_id(first_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::swap_sub_tree: Missing an error value on finding an invalid NodeId."));
+        }
+
+        let (is_valid, error) = self.is_valid_node_id(second_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::swap_sub_tree: Missing an error value on finding an invalid NodeId."));
+        }
+
+        let lower_upper_test = self.find_subtree_root_between_ids(first_id, second_id)
+        .map(|_| (first_id, second_id))
+        .or(
+            self.find_subtree_root_between_ids(second_id, first_id)
+            .map(|_| (second_id, first_id))
+        );
+
```

Comment:
Do you mind adding a hanging indent here?
I would prefer something like this:
```
let lower_upper_test = self.find_subtree_root_between_ids(first_id, second_id)
    .map(|_| (first_id, second_id))
    .or(
        self.find_subtree_root_between_ids(second_id, first_id)
        .map(|_| (second_id, first_id))
    );
```
I know this isn't documented anywhere, I promise I"ll get a rust-fmt style set up soon!  I just haven't had much time to get it set up lately.